### ISSUE TITLE
no users imported

### ISSUE THREAD

**Issue by RiccardoAmadi** [qlik-oss/nprinting-adsync#3]
The application ends with a success message:
```
Performing an incremental sync.
Incremental sync successfully done.
```
but no user is imported in NPrinting.
I don't receive any error message so I'm not sure where to search for further information. Is there a way to know if NPrinting connection of ADSync is working and to check the users found in AD by ADSync ?

### Versions

* NPrinting: 17.6
* Operating system: windows 2008 R2 Dataserver

EDIT:
I forced a full sync and i've got this as further response:

`Processing Group: NPtest_Users`

so ADSync is actually connected correctly to AD but then I get an error posting to nprinting:

```
API Error: Qlik.NPrinting.Rest.Client.Client.ApiException: Error calling UsersPo
st: {"error":{"code":400,"message":"Malformed parameters"}}
   at Qlik.NPrinting.Rest.Client.Api.UsersApi.UsersPostWithHttpInfo(UserRequest
userRequest)
   at ADSyncCSharp.Service.NpUserService.AddNewNpUser(User adUser)
   at ADSyncCSharp.Processor.ActiveDirectoryProcessor.HandleNPrintingUser(User u
ser)
   at ADSyncCSharp.Processor.ActiveDirectoryProcessor.ProcessNew(IList`1 entry)
   at ADSyncCSharp.Service.FullSyncOperation.execute(String querySearchFilter, I
EntryProcessor`1 entryProcessor)
   at ADSyncCSharp.Service.ActiveDirectorySyncService.FullSync()
   at ADSyncDemoApp.Component.AdSync.Sync()
   at ADSyncDemoApp.Program.Main(String[] args)
```

EDIT 2 [SOLVED]
I replaced the domain account which is optional in nprinting with an empty string and ADSync worked.
The replacement took place in NpUserService.cs on row 80.
The domain account string which was generating the malformed parameter exception contained a double backslash: `domain\\nptest.user1`

**Comment by vus-qlik** [comment]
Currently errors are only shown in the stdout. It seems there was nothing to process during incremental synch after changes made in AD.  Only users added to groups (defined in "searchFilter" query) will be taken into consideration during the synchronization process.

**Comment by RiccardoAmadi** [comment]
Right now ADSync is failing because I have a nested group. Can I ignore the inner group through the search filter or must I change the AD structure?

**Comment by vus-qlik** [comment]
Actually the searchFilter is just used to filter the groups that need to be replicated in NPrinting. The members of each group are then loaded through the C# GetMembers method (from the standard library GroupPrincipal) in FullSyncOperation and IncrementalSyncOperation which also fetch inner groups if any. I haven't investigated if it's possible to ignore inner groups through an ldap query but in any case code changes would be required to handle inner groups.

**Comment by RiccardoAmadi** [comment]
I managed to ignore the inner groups adding specific conditions on their CN in the search filter. I also added a filter to ignore users who have not an Email to avoid interrupting the whole script in that case and the import was partially successful. Now I'm struggling with a different problem: not all the users are imported, my guess is that users in a inner OU that aren't member of a group are not imported. Is that true? Is that a planned behaviour? I would like to understand if I should add every member to a group in AD or if I can workaround this somehow in ADSync. Thanks for your help
